### DISCUSSION TITLE
Highcharts stock chart dependency 404 error?

### DISCUSSION THREAD

**Discussion by ghost** [zauberzeug/nicegui#279] [Q&A] [answered]
I've been trying to implement the stock chart using[ this fiddle ](https://jsfiddle.net/gh/get/library/pure/highcharts/highcharts/tree/master/samples/stock/navigator/multiple-series/)from the highchart documentation. But I can't seem to get it to work. All it shows is the normal bar chart:

This is the code i'm using:
```py
from random import randint
from nicegui import ui

chart = ui.chart(extras=['stock'], 
                 options={
                    'chart': {
                        'type': 'line'
                    },
                     'rangeSelector': {
                        'selected': 4
                    },

                     'plotOptions': {
                         'series': {
                             'showInNavigator': "true"
                         }
                     },

                     'series': [{'data': [0, 2, 3, 6, 8, 10]}]
                 }).classes('w-full h-64')

def update():
    chart.options['series'][0]['data'].append(randint(0, 100))
    chart.update()

ui.button('Update', on_click=update)

ui.run()
```

The chrome dev tools seem to show that the stock dependency fails to import and can't be found:

> NiceGUI ready to go on http://0.0.0.0:8080
> http://127.0.0.1:8080/_nicegui/static/socket.io.min.js.map not found
> http://127.0.0.1:8080/_nicegui/dependencies/0/highcharts.js.map not found
> http://127.0.0.1:8080/_nicegui/dependencies/1/highcharts-more.js.map not found
> http://127.0.0.1:8080/_nicegui/dependencies/2/highcharts-3d.js.map not found
> http://127.0.0.1:8080/_nicegui/dependencies/52/stock.js.map not found

I am running nice gui 1.13 with a new clean venv of python 3.11. 
The [given example](https://nicegui.io/reference#chart) in the api reference works as expected for me.
Am I making a mistake in the implementation? Should I declare the type of chart somewhere, or is it something else I'm missing?

**Comment by falkoschindler** [comment, marked as answer]
Thanks for reaching out, @Mantanium!

First of all: The 404 errors are not related. When opening the developer console the browser tries to load *.js.map files, which are (currently) not provided by NiceGUI. We are still undecided if and how we would want to do that. For now, you can simply ignore these errors.

But you are expecting to see something like [this](https://jsfiddle.net/gh/get/library/pure/highcharts/highcharts/tree/master/samples/stock/navigator/multiple-series/) and not a plain line plot. I think the reason is that `ui.chart()` calls `Highcharts.chart()` and not `Highcharts.stockChart()`. Somehow we need to integrate this option into the API. I'll create an issue from this discussion: #284
